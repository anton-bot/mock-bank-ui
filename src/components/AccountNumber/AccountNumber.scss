.AccountNumber {
  font-weight: bold;
  display: flex;
  align-items: center;
}
