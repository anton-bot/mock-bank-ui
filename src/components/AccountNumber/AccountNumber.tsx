import { formatBankAccountNumber } from '../../functions/formatBankAccountNumber';
import './AccountNumber.scss';

type Props = {
  accountNumber: string;
};

export const AccountNumber: React.FC<Props> = (props) => (
  <div className="AccountNumber">{formatBankAccountNumber(props.accountNumber)}</div>
);
