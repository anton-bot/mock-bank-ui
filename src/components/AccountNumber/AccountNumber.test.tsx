import { render } from '@testing-library/react';
import { AccountNumber } from './AccountNumber';
import renderer from 'react-test-renderer';

const SAMPLE_ACCOUNT_NUMBER = '1234567890';

describe('Account Number', () => {
  it('renders bank account number', () => {
    const { getByText } = render(<AccountNumber accountNumber={SAMPLE_ACCOUNT_NUMBER} />);

    expect(getByText(/123-4567-890/)).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const tree = renderer.create(<AccountNumber accountNumber={SAMPLE_ACCOUNT_NUMBER} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
