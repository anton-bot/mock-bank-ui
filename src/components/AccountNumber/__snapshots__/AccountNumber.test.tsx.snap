// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account Number matches snapshot 1`] = `
<div
  className="AccountNumber"
>
  123-4567-890
</div>
`;
